drivers/sensor/stts22h/stts22h.c

```diff
@@ -0,0 +1,216 @@
+/* ST Microelectronics STTS22H temperature sensor
+ *
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/stts22h.pdf
+ */
+
+#define DT_DRV_COMPAT st_stts22h
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+
+#include "stts22h.h"
+
+LOG_MODULE_REGISTER(STTS22H, CONFIG_SENSOR_LOG_LEVEL);
+
+static inline int stts22h_set_odr_raw(const struct device *dev, stts22h_odr_temp_t odr)
+{
+	const struct stts22h_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+
+	return stts22h_temp_data_rate_set(ctx, odr);
+}
+
+static int stts22h_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct stts22h_data *data = dev->data;
+	const struct stts22h_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	int16_t raw_temp;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	if (stts22h_temperature_raw_get(ctx, &raw_temp) < 0) {
+		LOG_DBG("Failed to read sample");
+		return -EIO;
+	}
+
+	data->sample_temp = raw_temp;
+
+	return 0;
+}
+
+static inline void stts22h_temp_convert(struct sensor_value *val, int16_t raw_val)
+{
+	val->val1 = raw_val / 100;
+	val->val2 = ((int32_t)raw_val % 100) * 10000;
+}
+
+static int stts22h_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct stts22h_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_AMBIENT_TEMP) {
+		stts22h_temp_convert(val, data->sample_temp);
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const uint8_t stts22h_map[6] = {0, 1, 25, 50, 100, 200};
+
+static int stts22h_odr_set(const struct device *dev,
+			   const struct sensor_value *val)
+{
+	int odr, ret = 0;
+
+	for (odr = 0; odr < ARRAY_SIZE(stts22h_map); odr++) {
+		if (val->val1 <= stts22h_map[odr]) {
+			break;
+		}
+	}
+
+	switch (odr) {
+	case 0:
+		ret = stts22h_set_odr_raw(dev, STTS22H_POWER_DOWN);
+		break;
+	case 1:
+		ret = stts22h_set_odr_raw(dev, STTS22H_1Hz);
+		break;
+	case 2:
+		ret = stts22h_set_odr_raw(dev, STTS22H_25Hz);
+		break;
+	case 3:
+		ret = stts22h_set_odr_raw(dev, STTS22H_50Hz);
+		break;
+	case 4:
+		ret = stts22h_set_odr_raw(dev, STTS22H_100Hz);
+		break;
+	case 5:
+		ret = stts22h_set_odr_raw(dev, STTS22H_200Hz);
```

Comment:
how about converting these to a series of

`return = stts22h_set_odr_raw(dev, STTS22H_200Hz);`

then you can drop the break, ret and even the final return (the compiler should stay happy). Makes it a bit more compact, you did it in other functions and I think it's cool.